### PR TITLE
Load dashboards from disk into the database at regular interval

### LINKED ISSUE #9654: Alternative to [dashboard.json] & document breaking change
[dashboard.json] option was removed in v5. Implement alternative or document change or how wizzy can be used to solve the same problem.

### PR DESCRIPTION
**Pull request by bergquist**
This PR makes it possible to load dashboards from disk into the database.

You can configure what folders to scan in the `/conf/dashboards` folder. 
```yaml
- name: 'default'
  org_id: 1
  folder: ''
  type: file
  options:
    folder: /var/lib/grafana/dashboards
```

Grafana will scan those folders every 5s and see if the Modification time on the dashboard have changed. Updated dashboards will be loaded into the database. 

Dashboards loaded from disk will have `editable: false`. 

closes #9654 #5269

### CODE REVIEW

File: pkg/services/provisioning/dashboards/file_reader.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+type fileReader struct {
+	Cfg           *DashboardsAsConfig
+	Path          string
+	log           log.Logger
+	dashboardRepo dashboards.Repository
+	cache         *gocache.Cache
+}
+
+func NewDashboardFilereader(cfg *DashboardsAsConfig, log log.Logger) (*fileReader, error) {
```
Comment (torkelo) [inline pkg/services/provisioning/dashboards/file_reader.go]: nitpick: Capital R?

**Review by torkelo — COMMENTED** [review]
Looks very good! 

We need to think about how the dashboards path and datasources path (and example yaml file) will be for deb & rpm packages. the conf/ path is mostly used for local dev setups and windows setups. All linux setups have config path set to /etc/grafana  (via initd & systemd launch params). Should the deb & rpm post install steps copy an example file to /etc/grafana ? what should the default dashboards dir be? /var/lib/grafana/dashboards ? 

Second thing we need to think about is adding another option than editable. I do not think editable: true/false is good enough for this feature (at least not long term) as editable dashboards cannot be modified in the UI (event transiently, so you cannot go into edit mode in panels). So maybe we need an option like "changeFromUI" or something like that, so people can modify it from the UI but not save it (or get a warning when saving, like with app dashboards). But maybe editable false is ok, to update the dashboard they wil have to make it editable first, then modify it, then View JSON , then save to file. Not sure if that workflow is good enough

**Comment by bergquist** [comment]
`editable: false` was not intended to be a long-term solution. But I think its good enough as a [dashboard.json] replacement.  Playing around with dashboards from file should be a fairly limited edge case for advanced users. Those can make it editable for that purpose. Lets get back to this once the UX for 5.0 is more stable. 

Will solve the packing part.

File: packaging/deb/init.d/grafana-server [hunk 2]
```diff
@@ -33,6 +33,8 @@ DATA_DIR=/var/lib/grafana
 PLUGINS_DIR=/var/lib/grafana/plugins
 LOG_DIR=/var/log/grafana
 CONF_FILE=$CONF_DIR/grafana.ini
+DATASOURCES_CFG_DIR=$CONF_DIR/datasources
+DASHBOARDS_CFG_DIR=$CONF_DIR/dashboards
```
Comment (torkelo) [inline packaging/deb/init.d/grafana-server]: I wonder if we it's wise to have separate configs for this. Maybe we will add users and panels as well in the future and that will be a lot of path configs. Maybe we can combine these to a provisioning dir? and under that dir grafana will create (if not exists) a datasources and dashboards dirs. alternatively we add a config_dir option that works the same but just more general than less specific than conf_file

Comment (bergquist) [reply packaging/deb/init.d/grafana-server]: At first, I was hesitant. But I like this idea. Esp once we get beyond dahboards/datasources

How about:

[path]
~~dashboards~~ = 
~~datasources~~
provisioning = conf/provisioning


```bash
/etc/grafana/provisioning/dashboards
/etc/grafana/provisioning/datasources
/etc/grafana/provisioning/plugins
/etc/grafana/provisioning/alerts
/etc/grafana/provisioning/alert-notifiers
/etc/grafana/provisioning/template-vars
```

**Pull request merged**